lib/permutations.rb

```diff
@@ -1,4 +1,31 @@
 
 def permutations?(string1, string2)
+  # take two strings as arguments and returns true if one string is a permutation of the other
   
+  hash = {}
+  string1.split("").each do |char|
+    if hash[char]
+      hash[char] += 1
+    else
+      hash[char] = 1
+    end
+  end
+  
+  puts hash
+  string2.split("").each do |char|
+    if ! hash[char]
```

Comment:
For good style, the `!` should be immediately next to the value it is associated with. So in this case it would be
```ruby
if !hash[char]
```

Another way to write this without the bang (the `!`) would be:
```ruby
unless hash[char]
```